alerting/src/main/kotlin/org/opensearch/alerting/BucketLevelMonitorRunner.kt

```diff
@@ -325,11 +335,34 @@ object BucketLevelMonitorRunner : MonitorRunner() {
                         continue
 
                     val actionCtx = triggerCtx.copy(
-                        dedupedAlerts = dedupedAlerts,
+                        dedupedAlerts = dedupedAlerts.map {
+                            val dedupedAlertsComments = CommentsUtils.getCommentsForAlertNotification(
```

Comment:
As discussed offline, `getCommentsForAlertNotification` is being called for each alert in the `dedupedAlertsComments`, `newAlerts`, and `completedAlerts` lists. Since a notification from a bucket monitor can easily include dozens of alerts, that would mean dozens of search requests. @toepkerd is investigating consolidating the searchRequests for comments across all the lists into a single/batched request.